azkaban-exec-server/src/main/java/azkaban/container/FlowContainer.java

```diff
@@ -758,7 +761,10 @@ void shutdown() {
       logger.error("Error shutting down JettyServer while winding down the FlowContainer", e);
     }
     logger.info("Sayonara!");
-    uploadLogFile(execId);
+    if (this.flowRunner != null) {
```

Comment:
My understanding (which might be wrong), is that this is a change that is needed to quickly mitigate an issue. Additionally, my understanding of the Azkaban code is not the best. However, it seems like we should strive for a state where the flowRunner cannot be null, especially because we control the flowRunner object internally, right? Would it be wise to work toward such a state in the future?

Reply:
It is possible. What is happening is for some reason, the FlowRunner object is not created and shutdown is called (being investigated separately) which is causing this problem.
If it is identified that it is possible to not have FlowRunner object in some cases, we may have to keep this logic.